src/game/gameflow/gameflow_new.c

```diff
@@ -83,8 +83,137 @@ static void M_LoadGameStrings(const int32_t level_num)
 
 void GF_N_LoadStrings(const int32_t level_num)
 {
+    Object_ResetNames();
+
     M_LoadObjectStrings(level_num);
     M_LoadGameStrings(level_num);
+
+    struct {
+        GAME_OBJECT_ID object_id;
+        GF_GAME_STRING game_string;
+    } game_string_defs[] = {
+        { O_COMPASS_OPTION, GF_S_GAME_INV_ITEM_STATISTICS },
+        { O_COMPASS_ITEM, GF_S_GAME_INV_ITEM_STATISTICS },
+        { O_PISTOL_ITEM, GF_S_GAME_INV_ITEM_PISTOLS },
+        { O_PISTOL_OPTION, GF_S_GAME_INV_ITEM_PISTOLS },
+        { O_FLARE_ITEM, GF_S_GAME_INV_ITEM_FLARE },
```

Comment:
I think you forgot an entry for: `{ O_FLARE_OPTION, ...},` so it falls back to `OBJ_NAME_DEFINE(O_FLARES_OPTION, "Flare Option")` in libtrx `include/libtrx/game/objects/names_tr2.def`